### ISSUE TITLE
something wrong with login apple id

### ISSUE THREAD

**Issue by vancex** [fastlane/fastlane#4590]
### New Issue Checklist

- [ ] Updated fastlane to the latest version
- [ ] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description
I can't login into apple id now,  it lasts for hours
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/net/http/response.rb:357:in `finish': buffer error (Zlib::BufError)
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/net/http/response.rb:357:in `finish'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/net/http/response.rb:262:in `ensure in inflater'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/net/http/response.rb:262:in `inflater'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/net/http/response.rb:274:in `read_body_0'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/net/http/response.rb:201:in `read_body'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/net/http.rb:1127:in `block in get'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/net/http.rb:1413:in `block (2 levels) in transport_request'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/net/http/response.rb:162:in `reading_body'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/net/http.rb:1412:in `block in transport_request'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/net/http.rb:1403:in `catch'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/net/http.rb:1403:in `transport_request'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/net/http.rb:1376:in `request'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/net/http.rb:1369:in `block in request'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/net/http.rb:852:in `start'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/net/http.rb:1367:in `request'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/net/http.rb:1126:in `get'
	from /Library/Ruby/Gems/2.0.0/gems/faraday-0.9.2/lib/faraday/adapter/net_http.rb:80:in `perform_request'
	from /Library/Ruby/Gems/2.0.0/gems/faraday-0.9.2/lib/faraday/adapter/net_http.rb:40:in `block in call'
	from /Library/Ruby/Gems/2.0.0/gems/faraday-0.9.2/lib/faraday/adapter/net_http.rb:87:in `with_net_http_connection'
	from /Library/Ruby/Gems/2.0.0/gems/faraday-0.9.2/lib/faraday/adapter/net_http.rb:32:in `call'
	from /Library/Ruby/Gems/2.0.0/gems/faraday-cookie_jar-0.0.6/lib/faraday/cookie_jar.rb:22:in `call'
	from /Library/Ruby/Gems/2.0.0/gems/faraday_middleware-0.10.0/lib/faraday_middleware/response_middleware.rb:30:in `call'
	from /Library/Ruby/Gems/2.0.0/gems/faraday_middleware-0.10.0/lib/faraday_middleware/response_middleware.rb:30:in `call'
	from /Library/Ruby/Gems/2.0.0/gems/faraday_middleware-0.10.0/lib/faraday_middleware/response_middleware.rb:30:in `call'
	from /Library/Ruby/Gems/2.0.0/gems/faraday-0.9.2/lib/faraday/rack_builder.rb:139:in `build_response'
	from /Library/Ruby/Gems/2.0.0/gems/faraday-0.9.2/lib/faraday/connection.rb:377:in `run_request'
	from /Library/Ruby/Gems/2.0.0/gems/faraday-0.9.2/lib/faraday/connection.rb:140:in `get'
	from /Library/WebServer/certworkspace/api/spaceship/lib/spaceship/client.rb:404:in `block in send_request'
	from /Library/WebServer/certworkspace/api/spaceship/lib/spaceship/client.rb:322:in `with_retry'
	from /Library/WebServer/certworkspace/api/spaceship/lib/spaceship/client.rb:403:in `send_request'
	from /Library/WebServer/certworkspace/api/spaceship/lib/spaceship/client.rb:379:in `request'
	from /Library/WebServer/certworkspace/api/spaceship/lib/spaceship/client.rb:313:in `itc_service_key'
	from /Library/WebServer/certworkspace/api/spaceship/lib/spaceship/client.rb:274:in `block in send_shared_login_request'
	from /Library/Ruby/Gems/2.0.0/gems/faraday-0.9.2/lib/faraday/connection.rb:374:in `block in run_request'
	from /Library/Ruby/Gems/2.0.0/gems/faraday-0.9.2/lib/faraday/connection.rb:388:in `block in build_request'
	from /Library/Ruby/Gems/2.0.0/gems/faraday-0.9.2/lib/faraday/request.rb:26:in `block in create'
	from /Library/Ruby/Gems/2.0.0/gems/faraday-0.9.2/lib/faraday/request.rb:25:in `tap'
	from /Library/Ruby/Gems/2.0.0/gems/faraday-0.9.2/lib/faraday/request.rb:25:in `create'
	from /Library/Ruby/Gems/2.0.0/gems/faraday-0.9.2/lib/faraday/connection.rb:384:in `build_request'
	from /Library/Ruby/Gems/2.0.0/gems/faraday-0.9.2/lib/faraday/connection.rb:370:in `run_request'
	from /Library/Ruby/Gems/2.0.0/gems/faraday-0.9.2/lib/faraday/connection.rb:177:in `post'
	from /Library/WebServer/certworkspace/api/spaceship/lib/spaceship/client.rb:404:in `block in send_request'
	from /Library/WebServer/certworkspace/api/spaceship/lib/spaceship/client.rb:322:in `with_retry'
	from /Library/WebServer/certworkspace/api/spaceship/lib/spaceship/client.rb:403:in `send_request'
	from /Library/WebServer/certworkspace/api/spaceship/lib/spaceship/client.rb:379:in `request'
	from /Library/WebServer/certworkspace/api/spaceship/lib/spaceship/client.rb:273:in `send_shared_login_request'
	from /Library/WebServer/certworkspace/api/spaceship/lib/spaceship/portal/portal_client.rb:35:in `send_login_request'
	from /Library/WebServer/certworkspace/api/spaceship/lib/spaceship/client.rb:346:in `do_login'
	from /Library/WebServer/certworkspace/api/spaceship/lib/spaceship/client.rb:225:in `login'
	from /Library/WebServer/certworkspace/api/spaceship/lib/spaceship/client.rb:88:in `login'
	from /Library/WebServer/certworkspace/api/spaceship/lib/spaceship/portal/spaceship.rb:22:in `login'
	from /Library/WebServer/certworkspace/api/spaceship/lib/spaceship/portal/spaceship.rb:77:in `login'
	from /Library/WebServer/certworkspace/api/spaceship/bin/get_production_cert:43:in `<main>'

##### Complete output when running fastlane, including the stack trace and command used

```
INSERT OUTPUT HERE
here is the log file
1 # Logfile created on 2016-05-10 17:00:05 +0800 by logger.rb/41954
  2 [17:00:05]: >> POST:
  3 [17:00:05]: >> GET: https://itunesconnect.apple.com/itc/static-resources/controllers/login_cntrl.js
```

### Configuration Files

Please copy the complete content of your `Fastfile` and any other configuration files you use below:

Fastfile:
```ruby
# INSERT CONTENT HERE
```

### Environment

fastlane version (run `fastlane -v`):

Do you use bundler to execute fastlane (i.e. `bundle exec fastlane`)?

Do you use a Ruby environment manager (e.g. `chruby`, `rbenv`, `rvm`)?

**Comment by rizwankce** [comment]
I am also having the same issue. Report id:`59be9115b81676a433a3f3a4f91b0add`

**Comment by vancex** [comment]
@rizwankce 
what do u think causes this problem, something changed by the apple protocol side?

**Comment by vpolouchkine** [comment]
Can you provide repro steps for this? What command are you running?

**Comment by vancex** [comment]
just spaceship.login，I have found that something is wrong when  get https itunes.apple.com/xxxxx for getting itc service key，i can not remember how to spell it exactly。
i sloved it by hardcode a key，and next steps go fine

**Comment by vancex** [comment]
I dig the problem, it's happens when GET https://itunesconnect.apple.com/itc/static-resources/controllers/login_cntrl.js for itc_service_key,.
when request itc_service_key, it will causes the crash showed in the backtrace above, and I tried that not to get itc_service_key instead I write a key value down for the widgetKey, it goes fine now, but I don't understand when will the key changes

**Comment by rizwankce** [comment]
@vancex please let me know how can i fix this issue and login into iTunes connect?

**Comment by vancex** [comment]
@rizwankce leave me a email

**Comment by harryzhao2015** [comment]
i also have the same issue when run 'sigh' directly. report id: 5ac329096a857ff750fbee51d1b08563

**Comment by vpolouchkine** [comment]
Thanks for the reports, we'll take a look. 

@vancex What value are you hardcoding?

**Comment by vpolouchkine** [comment]
@vancex Also, do you have 2-factor auth enabled?

**Comment by mfurtak** [comment]
Hi @harryzhao2015 :wave: - we just released [`spaceship` 0.26.3](https://github.com/fastlane/fastlane/releases/tag/spaceship/0.26.3) 🚀  that includes an experimental workaround behind a feature switch.

Could you please upgrade your `spaceship` with:

```bash
gem update spaceship
```

and make sure that 

```bash
gem which spaceship
```

returns a path including `0.26.3`, and then try running:

```bash
SPACESHIP_LOGIN_ENCODING_IDENTITY=1 sigh [YOUR PARAMS]
```

and let me know if it allows you to succeed? Thank you very much!

**Comment by mfurtak** [comment]
Hi all, we've released [`spaceship` 0.27.1](https://github.com/fastlane/fastlane/releases/tag/spaceship/0.27.1) 🚀  This makes the fix provided by `SPACESHIP_LOGIN_ENCODING_IDENTITY` enabled by default, so if you upgrade, you should no longer need to use that environment variable.

Thanks for your help in investigating this issue! 👍
